apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs

```diff
@@ -413,26 +418,81 @@ private LogEntry BuildLogEntry(LogEventInfo loggingEvent)
         {
             var logEntry = new LogEntry
             {
-                TextPayload = RenderLogEvent(Layout, loggingEvent),
+
                 Severity = s_levelMap[loggingEvent.Level],
                 Timestamp = ConvertToTimestamp(loggingEvent.TimeStamp),
                 LogName = _logName,
                 Resource = _resource,
             };
 
-            foreach (var combinedProperty in GetAllProperties(loggingEvent))
+            if (SendJsonPayload)
             {
-                if (string.IsNullOrEmpty(combinedProperty.Key))
-                    continue;
+                var jsonStruct = new Struct();
+                jsonStruct.Fields.Add("message", Value.ForString(RenderLogEvent(Layout, loggingEvent)));
 
-                try
+                var propertiesStruct = new Struct();
+                jsonStruct.Fields.Add("properties", Value.ForStruct(propertiesStruct));
+
+                foreach (var combinedProperty in GetAllProperties(loggingEvent))
                 {
-                    logEntry.Labels[combinedProperty.Key] = combinedProperty.Value?.ToString() ?? "null";
+                    if (string.IsNullOrEmpty(combinedProperty.Key))
+                        continue;

```

Comment:
Please use braces around `continue`:
```C#
if (string.IsNullOrEmpty(combinedProperty.Key))
{
    continue;
}
```

I know this makes this if statement more verbose, but our coding standard is to always use braces.

Reply:
Fixed